test/e2e/utils/utils.go

```diff
@@ -149,3 +149,17 @@ func DeleteResource(t *testing.T, obj DeleteObjects) {
 		t.Fatalf("Failed to delete %s: %v", obj.GVR.Resource, err)
 	}
 }
+
+// CheckOperationAllowed checks whether operation are allowed
+func CheckOperationAllowed(eventMap map[utils.EventKind]bool, namespace string, resource string, eventType config.EventType) bool {
```

Comment:
@A-kanksh-a I meant to add this function in `pkg/utils` and use it in controller.go as well

Reply:
yes we can do that.